java-checks/src/main/java/org/sonar/java/checks/ThreadAsRunnableArgumentCheck.java

```diff
@@ -0,0 +1,118 @@
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02
+ */
+package org.sonar.java.checks;
+
+import com.google.common.collect.ImmutableList;
+import org.sonar.api.server.rule.RulesDefinition;
+import org.sonar.check.Priority;
+import org.sonar.check.Rule;
+import org.sonar.java.model.AbstractTypedTree;
+import org.sonar.java.model.expression.MethodInvocationTreeImpl;
+import org.sonar.java.model.expression.NewClassTreeImpl;
+import org.sonar.java.resolve.Symbol;
+import org.sonar.java.resolve.Symbol.MethodSymbol;
+import org.sonar.java.resolve.Type;
+import org.sonar.java.resolve.Type.ArrayType;
+import org.sonar.plugins.java.api.tree.ExpressionTree;
+import org.sonar.plugins.java.api.tree.IdentifierTree;
+import org.sonar.plugins.java.api.tree.Tree;
+import org.sonar.plugins.java.api.tree.Tree.Kind;
+import org.sonar.squidbridge.annotations.ActivatedByDefault;
+import org.sonar.squidbridge.annotations.SqaleConstantRemediation;
+import org.sonar.squidbridge.annotations.SqaleSubCharacteristic;
+
+import java.text.MessageFormat;
+import java.util.List;
+
+@Rule(
+  key = "S2438",
+  name = "\"Threads\" should not be used where \"Runnables\" are expected",
+  tags = {"multi-threading", "pitfall"},
+  priority = Priority.CRITICAL)
+@ActivatedByDefault
+@SqaleSubCharacteristic(RulesDefinition.SubCharacteristics.INSTRUCTION_RELIABILITY)
+@SqaleConstantRemediation("15min")
+public class ThreadAsRunnableArgumentCheck extends SubscriptionBaseVisitor {
+
+  @Override
+  public List<Kind> nodesToVisit() {
+    return ImmutableList.of(Kind.NEW_CLASS, Kind.METHOD_INVOCATION);
+  }
+
+  @Override
+  public void visitNode(Tree tree) {
+    if (!hasSemantic()) {
+      return;
+    }
+    List<ExpressionTree> arguments;
+    Symbol methodSymbol;
+    if (tree.is(Kind.NEW_CLASS)) {
+      NewClassTreeImpl nct = (NewClassTreeImpl) tree;
+      methodSymbol = getSemanticModel().getReference(nct.getConstructorIdentifier());
+      arguments = nct.arguments();
+    } else {
+      MethodInvocationTreeImpl mit = (MethodInvocationTreeImpl) tree;
+      methodSymbol = mit.getSymbol();
+      arguments = mit.arguments();
+    }
+    // FIXME semantic - symbol should never be null!
+    if (!arguments.isEmpty() && methodSymbol != null && methodSymbol.isKind(Symbol.MTH)) {
+      checkArgumentsTypes(arguments, (MethodSymbol) methodSymbol);
+    }
+  }
+
+  private void checkArgumentsTypes(List<ExpressionTree> arguments, MethodSymbol methodSymbol) {
+    List<Type> parametersTypes = methodSymbol.getParametersTypes();
+    // FIXME semantic - if there is arguments and method symbol has been resolved, the list of parameters should not be empty!
```

Comment:
Can you produce a case ? why could the parametersTypes be empty ? this should lead to opening a jira ticket.